extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/mpmetrics/ConcurrentGaugeImpl.java

```diff
@@ -14,7 +15,8 @@
     Gauge gauge;
 
     ConcurrentGaugeImpl register(MpMetadata metadata, MetricDescriptor metricInfo, MeterRegistry registry) {
-        gauge = io.micrometer.core.instrument.Gauge.builder(metricInfo.name(), longAdder::longValue)
+        ToDoubleFunction<LongAdder> function = LongAdder::doubleValue;
+        gauge = io.micrometer.core.instrument.Gauge.builder(metricInfo.name(), longAdder, function)
```

Comment:
You don't need the indirection.. 

        gauge = io.micrometer.core.instrument.Gauge.builder(metricInfo.name(), longAdder, LongAdder::doubleValue)